Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp

```diff
@@ -3329,6 +3330,67 @@ Optional<Color> Parser::parse_rgb_or_hsl_color(StringView function_name, Vector<
     return {};
 }
 
+Optional<EdgeRect> Parser::parse_rect(Vector<ComponentValue> const& component_values)
+{
+    Length params[4];
+    auto tokens = TokenStream { component_values };
+    tokens.skip_whitespace();
+
+    auto maybe_length_1 = parse_length(tokens.next_token().token());
+    if (!maybe_length_1.has_value())
+        return {};
+    params[0] = maybe_length_1.release_value();
+
+    tokens.skip_whitespace();
+    if (tokens.peek_token().is(Token::Type::Comma)) {
+        tokens.next_token();
+    }
+
+    tokens.skip_whitespace();
+    auto maybe_length_2 = parse_length(tokens.next_token().token());
+    if (!maybe_length_2.has_value())
+        return {};
+    params[1] = maybe_length_2.release_value();
+
+    tokens.skip_whitespace();
+    if (tokens.peek_token().is(Token::Type::Comma)) {
+        tokens.next_token();
+    }
+
+    tokens.skip_whitespace();
+    auto maybe_length_3 = parse_length(tokens.next_token().token());
+    if (!maybe_length_3.has_value())
+        return {};
+    params[2] = maybe_length_3.release_value();
+
+    tokens.skip_whitespace();
+    if (tokens.peek_token().is(Token::Type::Comma)) {
+        tokens.next_token();
+    }
+
+    tokens.skip_whitespace();
+    auto maybe_length_4 = parse_length(tokens.next_token().token());
+    if (!maybe_length_4.has_value())
+        return {};
+    params[3] = maybe_length_4.release_value();
+
+    return EdgeRect { params[0], params[1], params[2], params[3] };
+}
+
+RefPtr<StyleValue> Parser::parse_rect_value(ComponentValue const& component_value)
+{
+    if (component_value.is_function()) {
+        auto& function = component_value.function();
+        if (function.name().equals_ignoring_case("rect"sv)) {
+            auto rect = parse_rect(function.values());
```

Comment:
Not entirely sure why we need both `parse_rect()` and `parse_rect_value()`? Is `parse_rect()` going to be used elsewhere?